app/templating/view_context.py

```diff
@@ -86,17 +87,32 @@ def build_view_context_for_question(current_location, variables, rendered_block,
     return context
 
 
-def build_view_context_for_summary(metadata, answer_store, schema_context, variables, csrf_token):
-    rendered_schema_json = renderer.render(g.schema.json, **schema_context)
-    summary_rendered_context = build_summary_rendering_context(g.schema, rendered_schema_json['sections'],
-                                                               answer_store, metadata)
+def build_view_context_for_summary(metadata, answer_store, schema_context, block_type, variables, csrf_token,
+                                   current_location):
+    section_list = []
+    section_title = None
+
+    if block_type == 'Summary':
+        section_list = renderer.render(g.schema.json, **schema_context)['sections']
+
+    elif block_type == 'SectionSummary':
+        for section in g.schema.json['sections']:
+            group_ids = (group['id'] for group in section['groups'])
+            if current_location.group_id in group_ids:
+                section_list = [renderer.render(section, **schema_context)]
+                section_title = section['title']
+                break
+
+    summary_rendering_context = build_summary_rendering_context(g.schema, section_list, answer_store, metadata)
 
     context = {
         'csrf_token': csrf_token,
         'summary': {
-            'groups': summary_rendered_context,
+            'groups': summary_rendering_context,
             'answers_are_editable': True,
             'is_view_submission_response_enabled': is_view_submitted_response_enabled(g.schema.json),
+            'summary_type': block_type,
+            'section_title': section_title,
```

Comment:
This should be populated for both summary types and take out any if conditions in the html. So for the main summary the title would be "Your answers", for SectionSummary it would be the section title. Maybe look at renaming to something more specific to this use. Possibly just title.